dbms/programs/client/Client.cpp

```diff
@@ -602,6 +602,7 @@ class Client : public Poco::Util::Application
 
             if (!ends_with_backslash && (ends_with_semicolon || has_vertical_output_suffix || (!config().has("multiline") && !hasDataInSTDIN())))
             {
+                // TODO: should we do sensitive data masking on client too? History file can be source of secret leaks.
```

Comment:
It's questionable. Because the history file is stored in user home directory that is not accessible by other users.